dist/web3-light.js

```diff
@@ -3495,6 +3516,201 @@ module.exports = ICAP;
+
+    this.connection = net.connect({path: this.path});
+
+    this.connection.on('error', function(e){
+        console.error('IPC Connection Error', e);
+        _this._timeout();
+    });
+
+    this.connection.on('end', function(){
+        _this._timeout();
+    }); 
+
+
+    // LISTEN FOR CONNECTION RESPONSES
+    this.connection.on('data', function(data) {
+        /*jshint maxcomplexity: 6 */
+        data = data.toString();
+
+        // DE-CHUNKER
+        var dechunkedData = data
+            .replace(/\}\{/g,'}|--|{') // }{
+            .replace(/\}\]\[\{/g,'}]|--|[{') // }][{
+            .replace(/\}\[\{/g,'}|--|[{') // }[{
+            .replace(/\}\]\{/g,'}]|--|{') // }]{
+            .split('|--|');
+
+
+        dechunkedData.forEach(function(data){
+
+            // prepend the last chunk
+            if(_this.lastChunk)
+                data = _this.lastChunk + data;
+
+            var result = data,
+                id = null;
+
+            try {
+                result = JSON.parse(result);
+
+            } catch(e) {
+
+                _this.lastChunk = data;
+
+                // start timeout to cancel all requests
+                clearTimeout(_this.lastChunkTimeout);
+                _this.lastChunkTimeout = setTimeout(function(){
+                    _this.timeout();
+                    throw errors.InvalidResponse(result);        
+                }, 1000 * 15);
+
+                return;
+            }
+
+            // cancel timeout and set chunk to null
+            clearTimeout(_this.lastChunkTimeout);
+            _this.lastChunk = null;
+
+            // get the id which matches the returned id
+            if(utils.isArray(result)) {
+                result.forEach(function(load){
+                    if(_this.responseCallbacks[load.id])
+                        id = load.id;
+                });
+            } else {
+                id = result.id;
+            }
+
+            // fire the callback
+            if(_this.responseCallbacks[id]) {
+                _this.responseCallbacks[id](null, result);
+                delete _this.responseCallbacks[id];
+            }
+        });
+    });
+};
+
+/**
+Get the adds a callback to the responseCallbacks object,
+which will be called if a response matching the response Id will arrive.
+
+@method _getResponse
+*/
+IpcProvider.prototype._getResponse = function(payload, callback) {
+    var id = payload.id || payload[0].id;
+    var method = payload.method || payload[0].method;
+
+    this.responseCallbacks[id] = callback;
+    this.responseCallbacks[id].method = method;
+};
+
+/**
+Timeout all requests when the end/error event is fired
+
+@method _timeout
+*/
+IpcProvider.prototype._timeout = function() {
+    for(var key in this.responseCallbacks) {
+        if(this.responseCallbacks.hasOwnProperty(key)){
+            this.responseCallbacks[key](errorTimeout.replace('__id__', key).replace('__method__', this.responseCallbacks[key].method));
```

Comment:
no sure if that makes sense, as its the only time its done here.
Reducing cyclomatic complexity this way is fake reducing it, as it still has to execute the same amount of complexity, just jslint doesn't see it.